### PR TITLE
Always go through cmake/ChooseBlas.cmake, now we only execute this lo…

### PR DESCRIPTION
**Pull request by larroy**
…gic when using CUDA

Blas logic should be independent of CUDA

## Description ##
see above

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -294,8 +294,8 @@ else()
   add_definitions(-DMXNET_USE_NCCL=0)
 endif()
 
+include(cmake/ChooseBlas.cmake)
```
Comment (lebeg) [inline CMakeLists.txt]: Lots of stuff from here will be overwritten (and misconfigured) due to the `else()` branch, when the [mshadow scripts](https://github.com/dmlc/mshadow/blob/master/cmake/mshadow.cmake) kick in. On the other hand if it passes CI, then why not?

Comment (larroy) [reply CMakeLists.txt]: Well, I need the defines for the feature detection basically.

Comment (edisongustavo) [reply CMakeLists.txt]: Yes, as @lebeg mentioned this change will do nothing in the current state of the code, so this PR doesn't bring anything to the table.

I am currently working on making linking with BLAS a little bit easier, specially on Windows.

**Comment by vandanavk** [comment]
@larroy could you re-trigger the CI?

File: CMakeLists.txt (hunk 1 above)
Comment (larroy) [reply CMakeLists.txt]: It works for the feature detection though

**Pull request closed without merging**